types/types.go

```diff
@@ -198,6 +199,42 @@ type NodeConfig struct {
 	// Introduced to prevent the check from running with ext-containers, since
 	// they should be present by definition.
 	SkipUniquenessCheck bool
+	Components          []*Component
+}
+
+func (n *NodeConfig) Copy() *NodeConfig {
+	if n == nil {
+		return nil
+	}
+
+	copyConfig := *n
+
+	// Deep copy maps
+	copyConfig.Sysctls = utils.CopyMap(n.Sysctls)
+	copyConfig.Env = utils.CopyMap(n.Env)
+	copyConfig.Labels = utils.CopyMap(n.Labels)
```

Comment:
@steiler shall we consider using either one of the deepcopy packages or try unmarshal/marshal approach - https://goplay.tools/snippet/zmnfT77uIQa